kuka_rsi_simulator/kuka_rsi_simulator/simulator.py

```diff
@@ -0,0 +1,94 @@
+import errno
+import socket
+import sys
+import time
+import xml.etree.ElementTree as ET
+
+import numpy as np
+import rclpy
+from rclpy.node import Node
+
+
+class RsiSimulator(Node):
+    def __init__(self, node_name: str, host: str, port: int):
+        super().__init__(node_name)
+        self._cycle_time = 0.004
+        self._host = host
+        self._port = port
+        self._q = np.array([0, -90, 90, 0, 90, 0]).astype(np.float64)
+        self._timeout_count = 0
+        self._ipoc = 0
+        self._timer = self.create_timer(self._cycle_time, self.timer_callback)
+
+        try:
+            self._s = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+            self.get_logger().info("Successfully created socket")
+            self._s.settimeout(1)
+        except socket.error as e:
+            self.get_logger().fatal(f"Could not create socket: {e}")
+            sys.exit()
+
+    def close_socket(self):
+        self._s.close()
+
+    def _create_rsi_xml_rob(self):
+        root = ET.Element('Rob', {'TYPE': 'KUKA'})
+        ET.SubElement(root, 'RIst', {'X': '0.0', 'Y': '0.0', 'Z': '0.0',
+                                     'A': '0.0', 'B': '0.0', 'C': '0.0'})
+        ET.SubElement(root, 'AIPos', {'A1': str(self._q[0]), 'A2': str(self._q[1]), 'A3': str(self._q[2]),
+                                      'A4': str(self._q[3]), 'A5': str(self._q[4]), 'A6': str(self._q[5])})
+        ET.SubElement(root, 'Delay', {'D': str(self._timeout_count)})
+        ET.SubElement(root, 'IPOC').text = str(self._ipoc)
+        return ET.tostring(root)
+
+    def _parse_rsi_xml_sen(self, data):
+        root = ET.fromstring(data)
+        axis = root.find('Axis').attrib
+        desired_joint_correction = np.array([axis['A1'], axis['A2'], axis['A3'],
+                                             axis['A4'], axis['A5'], axis['A6']]).astype(np.float64)
+        if self._ipoc != int(root.find('IPOC').text):
+            raise
+        return desired_joint_correction
+
+    def timer_callback(self):
+        try:
+            msg = self._create_rsi_xml_rob()
+            self._s.sendto(msg, (self._host, self._port))
+            recv_msg, addr = self._s.recvfrom(1024)
+            des_joint_correction = self._parse_rsi_xml_sen(recv_msg)
+            self._q += 50 * des_joint_correction * self._cycle_time
```

Comment:
Because we are already sending a displacement (a relative correction) we don't actually want to integrate the velocity.  
```suggestion
            self._q += des_joint_correction
```